File: Server/microservices/task-service/src/main/java/com/taskman/task_service/service/interfaces/TaskService.java
package com.taskman.task_service.service.interfaces;

import com.taskman.task_service.dto.TaskDTO;
import com.taskman.task_service.dto.TaskAssignmentDTO;
import com.taskman.task_service.dto.request.CreateTaskRequest;
import com.taskman.task_service.dto.request.UpdateTaskRequest;
import com.taskman.task_service.entity.enums.TaskStatus;

import java.util.List;

public interface TaskService {
    // Task CRUD operations
    TaskDTO createTask(CreateTaskRequest request, String userId);
    TaskDTO getTaskById(Long id);
    List<TaskDTO> getAllTasks();
    TaskDTO updateTask(Long id, UpdateTaskRequest request, String userId);
    void deleteTask(Long id, String userId);

    // Task search and filter operations
    List<TaskDTO> findTasksByProjectId(String projectId);
    List<TaskDTO> findTasksByStatus(TaskStatus status);
    List<TaskDTO> findTasksByUserId(String userId);
    List<TaskDTO> findTasksByProjectIdAndStatus(String projectId, TaskStatus status);
    List<TaskDTO> findTasksByUserIdAndStatus(String userId, TaskStatus status);

    // Task assignment operations
    TaskAssignmentDTO assignTask(Long taskId, String assignerId, String assigneeId);
    void unassignTask(Long taskId, String userId, String assigneeId);
    List<TaskAssignmentDTO> getTaskAssignments(Long taskId);
    
    // Task status operations
    TaskDTO updateTaskStatus(Long taskId, TaskStatus newStatus, String userId);
    
    // Validation operations
    boolean isUserAssignedToTask(Long taskId, String userId);
    boolean canUserModifyTask(Long taskId, String userId);
} 

File: taskman_front/src/services/TaskService.js
import api from '../utils/axios';

export const TaskService = {
  async getAllTasks(projectId) {
    //http://localhost:8080/api/tasks?projectId=8
    const response = await api.get(`/tasks?projectId=${projectId}`);
    return response.data;
  },

  async getTaskById(taskId) {
    const response = await api.get(`/tasks/${taskId}`);
    return response.data;
  },

  async createTask(taskData) {
    const response = await api.post('/tasks', taskData);
    return response.data;
  },

  async updateTask(taskId, taskData) {
    const response = await api.put(`/tasks/${taskId}`, taskData);
    return response.data;
  },

  async updateTaskStatus(taskId, status) {
    const response = await api.put(`/tasks/${taskId}`, { status });
    return response.data;
  },

  async deleteTask(taskId) {
    const response = await api.delete(`/tasks/${taskId}`);
    return response.data;
  }
};

export default TaskService; 